### PR TITLE
adding logstash helm chart

### LINKED ISSUE #1591: logstash helmchart

### PR DESCRIPTION
**Pull request by dasrecht**
closes #1591

### CODE REVIEW

File: images/kubectl-build-deploy-dind/helmcharts/logstash/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+{{- define "logstash.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "logstash.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end -}}
+
+{{/*
+Create a PriorityClassName.
+(this is based on the Lagoon Environment Type)).
+*/}}
+{{- define "nginx-php.lagoon-priority" -}}
```
Comment (tobybellwood) [inline images/kubectl-build-deploy-dind/helmcharts/logstash/templates/_helpers.tpl]:
```suggestion
{{- define "logstash.lagoon-priority" -}}
```

File: images/kubectl-build-deploy-dind/helmcharts/logstash/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image }}"
```
Comment (tobybellwood) [inline images/kubectl-build-deploy-dind/helmcharts/logstash/templates/deployment.yaml]:
```suggestion
          image: {{ .Values.image | quote }}
```
Also, we've been placing this line before the name under containers:
image
name
securityContext
imagePullPolicy

File: images/kubectl-build-deploy-dind/helmcharts/logstash/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          imagePullPolicy: {{ .Values.imagePullPolicy }}
+          env:
+            - name: CRONJOBS
```
Comment (tobybellwood) [inline images/kubectl-build-deploy-dind/helmcharts/logstash/templates/deployment.yaml]: no cronjobs in logstash

File: images/kubectl-build-deploy-dind/helmcharts/logstash/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+              value: {{ .Values.cronjobs | quote }}
+            - name: LAGOON_GIT_SHA
+              value: {{ .Values.gitSha | quote }}
+          envFrom:
```
Comment (tobybellwood) [inline images/kubectl-build-deploy-dind/helmcharts/logstash/templates/deployment.yaml:34]: add SERVICE_NAME
```
            - name: SERVICE_NAME
              value: {{ .Release.Name | quote }}
```

File: images/kubectl-build-deploy-dind/helmcharts/logstash/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                name: lagoon-env
+          ports:
+            - name: 9600-tcp
+              containerPort: 9600
+              protocol: TCP
+          livenessProbe:
+            tcpSocket:
+              port: 9600
+            initialDelaySeconds: 15
+            timeoutSeconds: 1
```
Comment (tobybellwood) [inline images/kubectl-build-deploy-dind/helmcharts/logstash/templates/deployment.yaml:45]: liveness should have a periodSeconds

File: images/kubectl-build-deploy-dind/helmcharts/logstash/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+          readinessProbe:
+            tcpSocket:
+              port: 9600
+            initialDelaySeconds: 120
+            periodSeconds: 10
```
Comment (tobybellwood) [inline images/kubectl-build-deploy-dind/helmcharts/logstash/templates/deployment.yaml:52]: readinessProbe has a timeoutSeconds

File: images/kubectl-build-deploy-dind/helmcharts/logstash/templates/deployment.yaml (hunk 2 above)
Comment (dasrecht) [reply images/kubectl-build-deploy-dind/helmcharts/logstash/templates/deployment.yaml]: done

**Comment by dasrecht** [comment]
all done :)

**Pull request merged**